tracer/src/Datadog.Trace/Ci/ImpactedTestsModule.cs

```diff
@@ -0,0 +1,179 @@
+                        var modifiedFileBitmap = new FileBitmap(modifiedFile.ExecutedBitmap);
+
+                        if (testFileBitmap.IntersectsWith(ref modifiedFileBitmap))
+                        {
+                            Log.Debug("  Intersecting lines. Marking test {TestName} as modified.", test.Name);
+                            modified = true;
+                            break;
+                        }
+                    }
+                }
+
+                if (modified)
+                {
+                    tags.IsModified = "true";
+                    TelemetryFactory.Metrics.RecordCountCIVisibilityImpactedTestsIsModified();
+                }
+            }
+            else
+            {
+                Log.Debug("Impacted Tests Detection is DISABLED for {TestName}", test.Name);
+            }
+        }
+        catch (Exception err)
+        {
+            Log.Error(err, "Error analyzing Impacted Tests for {TestName}", tags.Name);
+        }
+    }
+
+    private static FileCoverageInfo[] GetTestCoverage(TestSpanTags tags)
+    {
+        if (tags.SourceFile is null || tags.SourceStart is null || tags.SourceEnd is null)
+        {
+            Log.Warning("No test definition file found for {TestName}", tags.Name);
+            return Array.Empty<FileCoverageInfo>();
+        }
+
+        // Milestone 1 : Return only the test definition file
+        var file = new FileCoverageInfo(tags.SourceFile);
+
+        // Milestone 1.5 : Return the test definition lines
+        if (CIEnvironmentValues.Instance.PrBaseCommit is { } prBase)
+        {
+            var executedBitmap = new FileBitmap((int)tags.SourceStart, (int)tags.SourceEnd);
+            file.ExecutedBitmap = executedBitmap.GetInternalArrayOrToArrayAndDispose();
+            Log.Debug<string, int, int>("TestCoverage for {TestFile}: {Start}..{End}", tags.SourceFile, (int)tags.SourceStart, (int)tags.SourceEnd);
+        }
+
+        return [file];
+    }
+
+    private static FileCoverageInfo[] GetModifiedFiles()
+    {
+        if (modifiedFiles is null)
+        {
+            lock (Log)
+            {
+                if (modifiedFiles is null)
+                {
+                    var workspacePath = CIEnvironmentValues.Instance.WorkspacePath ?? string.Empty;
+                    var prBase = CIEnvironmentValues.Instance.PrBaseCommit ?? GetBaseCommitFromBackend();
+                    var commit = CIEnvironmentValues.Instance.HeadCommit;
+                    if (prBase is { Length: > 0 })
+                    {
+                        Log.Debug("PR detected. Retrieving diff lines from Git CLI for {Path} {BaseCommit}...", workspacePath, prBase);
+                        // Milestone 1.5 : Retrieve diff files and lines from Git Diff CLI
+                        try
+                        {
+                            modifiedFiles = GitCommandHelper.GetGitDiffFilesAndLines(workspacePath, prBase, commit).Result;
```

Comment:
please don't do `.Result` over the async task

Reply:
Fixed